File: src/main/java/be/kdg/kandoe/backend/model/sessions/ParticipantInfo.java
package be.kdg.kandoe.backend.model.sessions;

import be.kdg.kandoe.backend.model.users.User;
import lombok.*;

import javax.persistence.*;

/**
 * This class holds information about the status of a participant of a session.
 */
@Data
@Entity
public class ParticipantInfo {
    @Id
    @GeneratedValue
    @Setter(AccessLevel.NONE)
    private int participantInfoId;
    
    @ManyToOne(targetEntity = User.class, optional = false, cascade = {CascadeType.REFRESH, CascadeType.MERGE}, fetch = FetchType.EAGER)
    private User participant;

    private int joinNumber;
    private boolean joined;
    private boolean addedCardsCompleted;
    private boolean reviewingCardsCompleted;

}

File: src/main/webapp/app/entities/session/participant-info.ts
import {Serializable} from "../../util/serializable";
import {User} from "../user/user";

/**
 * entity that represents a participantinfo JSON object
 */


export class ParticipantInfo implements Serializable<ParticipantInfo> {
    participant : User;
    joined : boolean;
    addedCardsCompleted : boolean;
    reviewingCardsCompleted : boolean;
    
    deserialize(object : ParticipantInfo) : ParticipantInfo {
        this.participant = User.createEmptyUser().deserialize(object.participant);
        this.joined = object.joined;
        this.addedCardsCompleted = object.addedCardsCompleted;
        this.reviewingCardsCompleted = object.reviewingCardsCompleted;
        
        return this;
    }
}
